pkg/controller/inferenceservice/resources/istio/virtualservice_test.go

```diff
@@ -60,6 +64,24 @@ func TestCreateVirtualService(t *testing.T) {
 			Gateways: []string{constants.KnativeLocalGateway},
 		},
 	}
+	predictorClusterLocalRouteMatch := []*istiov1alpha3.HTTPMatchRequest{
+		{
+			Authority: &istiov1alpha3.StringMatch{
+				MatchType: &istiov1alpha3.StringMatch_Regex{
+					Regex: constants.HostRegExp(constants.InferenceServiceHostName(serviceName, namespace, clusterLocalDomain)),
+				},
+			},
+			Gateways: []string{constants.KnativeIngressGateway},
```

Comment:
I think we need to remove the ingress gateway route match if it is configured with cluster local

Reply:
This could be another issue because both KNative gateways (ingress and local) are added by default in [virtualservice.go#L240-L257](https://github.com/kubeflow/kfserving/blob/master/pkg/controller/inferenceservice/resources/istio/virtualservice.go#L240-L257).